src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -148,12 +148,28 @@ public static void AssertValidVariableValue(ISchema schema, IGraphType type, obj
 
                 if (input is IValue value)
                 {
-                    if (scalar.ParseLiteral(value) == null)
+                    object convertedValue = null;
+                    try
+                    {
+                        convertedValue = scalar.ParseLiteral(value);
+                    }
+                    catch
+                    {
+                    }
+                    if (convertedValue == null)
                         throw new InvalidValueException(variableName, $"Unable to convert '{value.Value}' to '{type.Name}'");
                 }
                 else
                 {
-                    if (scalar.ParseValue(input) == null)
+                    object convertedValue = null;
+                    try
+                    {
+                        convertedValue = scalar.ParseValue(input);
+                    }
+                    catch
```

Comment:
Let's use thrown exception as inner exception for InvalidValueException.

Reply:
Ok but I want to be sure that `Data` from an inner exception does not get exposed outside the project, especially here where it is a query error (bad data), and not an unhandled exception.  Also, ExceptionError does not properly allow for setting the code to override the default.  I will make a separate PR to fix these issues, and then come back to this PR.

Reply:
Actually I'll just skip the code "issue", and throw the new constructor in this PR.

Reply:
If you are going to fix something in other PRs, temporary bypassing review remarks leaving such places unchanged, then I would really like to have the corresponding `//TODO: ` comments. I do not mind promoting the master branch, but only in such a way as to keep pending tasks in front of my eyes. Ok?

Reply:
I'm happy to oblige but want to make sure I understand you correctly.  You are saying that if you make a comment, and I suggest we fix in another PR, then I should add a comment to the code that says `//TODO:` -- for example, where you suggested that we unify the single and double quotation marks?  I'm not sure what you mean about promoting the master branch.

Reply:
(Usually I try to create a new issue in these cases, but am not always diligent in that.)

Reply:
I am not a native speaker. I mean I am not agains pushing in master PRs with some pending little issues not directly related to PR. Just marking these places in code with `// TODO: ` please, so as not to forget. Or making a new issue, yes. In case of `InvalidValueException` some data was lost. I always pay attention to such changes. My suggestion about unifying quotations is a different thing. This was just an abstract remark.

Reply:
I understand - thanks for the clarification!